packages/react/src/screens/auth-blocks/login-init/LoginInit.tsx

```diff
@@ -2,96 +2,137 @@ import type { LoginInitBlock, TextFieldWithError } from '@corbado/shared-ui';
 import React, { type FormEvent, useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 
-import { FormInput, Header, PrimaryButton, SubHeader } from '../../../components';
+import { PrimaryButton } from '../../../components/ui2/buttons/PrimaryButton';
+import { SecondaryButton } from '../../../components/ui2/buttons/SecondaryButton';
+import InputField from '../../../components/ui2/input/InputField';
+import { Header } from '../../../components/ui2/typography/Header';
+import { SubHeader } from '../../../components/ui2/typography/SubHeader';
+import { Text } from '../../../components/ui2/typography/Text';
 
 export const LoginInit = ({ block }: { block: LoginInitBlock }) => {
   const { t } = useTranslation('translation', { keyPrefix: `login.login-init.login-init` });
   const [loading, setLoading] = useState<boolean>(false);
-
-  const [emailOrUsername, setEmailOrUsername] = useState<TextFieldWithError | null>(null);
-  const emailorUsernameRef = useRef<HTMLInputElement>();
-
-  const [phone, setPhone] = useState<TextFieldWithError | null>(null);
-  const phoneRef = useRef<HTMLInputElement>();
+  const [textField, setTextField] = useState<TextFieldWithError | null>(null);
+  const [usePhone, setUsePhone] = useState<boolean>(block.data.isPhoneFocused || !block.data.emailOrUsernameEnabled);
+  const textFieldRef = useRef<HTMLInputElement>();
 
   useEffect(() => {
     setLoading(false);
     // TODO: set initial value of text field if available through block.data.loginIdentifier (this is important for aborted processes)
```

Comment:
Feel free to remove the TODO now (it was just a reminder that we need to do prefilling here)